app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,166 @@
+            negativeButtonText = negativeButtonText,
+            onPositiveBtnClick = onPositiveBtnClick,
+            onNegativeBtnClick = onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View,
+        cancelable: Boolean
+    ) {
+        createAndShowDialogSafely(
+            activity = activity,
+            title = title,
+            message = message,
+            positiveButtonText = activity.getString(R.string.yes),
+            negativeButtonText = activity.getString(R.string.no),
+            onPositiveBtnClick = onPositiveBtnClick,
+            onNegativeBtnClick = onNegativeBtnClick,
+            customView = customView,
+            cancelable = cancelable
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String?,
+        onPositiveBtnClick: Runnable?,
+        cancelable: Boolean
+    ) {
+        createAndShowDialogSafely(
+            activity = activity,
+            title = title,
+            message = message,
+            positiveButtonText = positiveButtonText,
+            onPositiveBtnClick = onPositiveBtnClick,
+            cancelable = cancelable
+        )
+    }
+
+    /**
+     * show a dialog
+     * @param activity
+     * @param title
+     * @param message
+     * @param positiveButtonText
+     * @param negativeButtonText
+     * @param onPositiveBtnClick
+     * @param onNegativeBtnClick
+     * @param customView
+     * @param cancelable
+     */
+    private fun createAndShowDialogSafely(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String? = null,
+        negativeButtonText: String? = null,
+        onPositiveBtnClick: Runnable? = null,
+        onNegativeBtnClick: Runnable? = null,
+        customView: View? = null,
+        cancelable: Boolean = false
+    ) {
+
+        /* If the custom view already has a parent, there is already a dialog showing with the view
+         * This happens for on resume - return to avoid creating a second dialog - the first one
+         * will still show
+         */
+        if (customView?.parent != null) {
+            return
+        }
+
+        showSafely(activity, AlertDialog.Builder(activity).apply {
+            setTitle(title)
+            setMessage(message)
+            setView(customView)
+            setCancelable(cancelable)
+            positiveButtonText?.let {
+                setPositiveButton(it) { _: DialogInterface, _: Int ->
+                    onPositiveBtnClick?.run()
+                }
+            }
+            negativeButtonText?.let {
+                setNegativeButton(it) { _: DialogInterface, _: Int ->
```

Comment:
we can fit these clicklisteners on one line if your remove the types from the lambda eg
```
setNegativeButton(it) { _, _ -> onNegativeBtnClick?.run() }
```

Reply:
Done!